### DISCUSSION TITLE
Apex Jorje seems to use insecure dependencies. What should we do?

### DISCUSSION THREAD

**Discussion by rsoesemann** [pmd/pmd#3690] [Q&A] [answered]
@adangel I am running snyk on my private PMD fork and it reported two violations in the Apex Jorje Parser

- https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLECODEGSON-1730327
- https://snyk.io/vuln/maven%3Ach.qos.logback%3Alogback-core

Both libs are used by Apex PMD Parser Jorje https://github.com/pmd/pmd/blob/pmd_releases/6.38.0/pmd-apex-jorje/pom.xml

**Comment by adangel** [comment, marked as answer]
We already do mangle the blob which contains apex jorje to make it smaller and avoid having dependencies more than once on the classpath: https://github.com/pmd/pmd/blob/master/pmd-apex-jorje/repo/create-local-repo.sh

It seems, we already remove everything except "com.google.common" (and the apex stuff, of course). Which means, we have the dependencies declared again explicitly in the  [pom.xml](https://github.com/pmd/pmd/blob/master/pmd-apex-jorje/pom.xml) and are actually able to change it.

I think, I used the same versions that apex jorje used when I created the script back then - so apex might have updated their dependencies as well already.

Given that the new versions are compatible, we could simply try to update. The tests should be able to tell, whether something breaks or whether everything still works as it should.

* com.google.code.gson:gson would be updated from 2.7 to 2.8.9 (or higher)
* ch.qos.logback:logback-core : 1.2.3 -> 1.2.7 (or higher)

I see, that we use gson 2.8.5 in pmd-core - I wonder, which one ends up actually in pmd-dist... it's actually 2.8.5 (which should be updated then as well).

Overall it seems to be doable.

**Reply by adangel** [reply]
See #3695 - the dependencies will be updated with 6.42.0
